addons/mail/models/mail_channel.py

```diff
@@ -149,25 +267,45 @@ def unlink(self):
 
     @api.multi
     def write(self, vals):
+        # First checks if user tries to modify moderation fields and has not the right to do it.
+        # We don't check if the fields touched are actually modified.
+        moderation_keys_are_touched = any(key for key in moderation_keys if vals.get(key))
+        if moderation_keys_are_touched and not self.env.user.has_group('base.group_system'):
+            if any(self.env.user not in channel.moderator_ids for channel in self if channel.moderation):
+                raise UserError("You do not possess the rights to modify fields related to moderation on one of the channels you are trying to modify!")
+
         tools.image_resize_images(vals)
         result = super(Channel, self).write(vals)
+
         if vals.get('group_ids'):
             self._subscribe_users()
+
+        if vals.get('moderation') is False:
+            MessagesToAccept = self.env['mail.message'].search(
```

Comment:
We limitate use of CamelCase to class. Here it is a record set, so let us just name it messages_to_accept. Or pending_messages.

Reply:
done